base/libgit2/callbacks.jl

```diff
@@ -110,13 +128,16 @@ function authenticate_ssh(creds::SSHCredentials, libgit2credptr::Ptr{Ptr{Void}},
             passdef = creds.pass # check if credentials were already used
             if passphrase_required && (isempty(passdef) || isusedcreds)
                 if Sys.iswindows()
-                    passdef = Base.winprompt(
+                    response = Base.winprompt(
                         "Your SSH Key requires a password, please enter it now:",
                         "Passphrase required", privatekey; prompt_username = false)
-                    isnull(passdef) && return Cint(Error.EAUTH)
-                    passdef = Base.get(passdef)[2]
+                    isnull(response) && return user_abort()
+                    passdef = unsafe_get(response)[2]
                 else
-                    passdef = prompt("Passphrase for $privatekey", password=true)
+                    response = prompt("Passphrase for $privatekey", password=true)
+                    isnull(response) && return user_abort()
+                    passdef = unsafe_get(response)
+                    isempty(passdef) && return user_abort()  # Ambiguous if EOF or newline
```

Comment:
When using `ssh-keygen` a passwordless key wouldn't pass the `passphrase_required` check.

For username/password I don't think LibGit2 will hit this callback unless it requires a password. I'll try to come up with an example for this.